README.md

```diff
@@ -1,90 +1,71 @@
-reviewing Vite's documentation on this here: [Vite Library Mode](https://vitejs.dev/guide/build.html#library-mode).
-
-In case you add direct or peer dependencies, you should uncomment following lines (and corresponding import above) to
-ensure those dependencies are not directly included in the built package.
-
-```tsx
-external: Object.keys(packageJson.dependencies).flatMap((dep) => [
-	dep,
-	// Include all dependency paths, not just root
-	new RegExp(`^${dep}/`),
-])
+```bash
+npm install @lokalise/frontegg-oauth-client
 ```
 
-## commitlint
-
-You can use `npm run commit` to interactively construct correct commit messsage.
-
-Check out [commitlint](https://commitlint.js.org) docs for examples of how to customise.
-
-## Release actions
-
-The following token needs to be set in the Github repo for the `prerelease` and `release` Github Actions to work:
-
-- `secrets.NPM_TOKEN` (need this to publish on NPM)
-
-When performing a release, make sure to follow our conventional commit approach, as described in [contribution documentation](https://github.com/lokalise/npm-package-template/blob/main/CONTRIBUTING.md).
-
-## License
-
-This project is APACHE, VERSION 2.0 licensed, see LICENSE.md for details.
-
-## Template Specific Internal Notes
-
-The following notes are only relevant to the maintainers of this template. You may delete this section when you clone it.
-
-### Creating a release of the template
-
-- Merge your PR and create a [new draft release](https://github.com/lokalise/npm-package-template/releases/new).
-- Create a new tag `vx.x.x`. Consider the type of changes you added. Major, minor or patch.
-- In Release title mention the same version `vx.x.x`.
-- Generate release notes.
-- Publish Release
-
-## Other Resources
+### Usage
+
+The library offers several APIs to initiate the OAuth flow and retrieve user data.
+
+```js
+// Create client instance
+const client = new FronteggOAuthClient({
+    baseUrl: 'https://frontegg-custom-url.com',
+    clientId: 'CLIENT_ID',
+    redirectUri: `${window.location.origin}/oauth/callback`,
+    logoutRedirectUri: window.location.origin,
+})
+
+try {
+    // Retrieve user
+    const user = await client.getUserData();
+
+    // Retrieve token 
+    const accessToken = user.accessToken;
+} catch (error) {
+    // In case we receive a 401 Unauthorized error, we need to redirect the user to the login page.
+    if (error instanceof FronteggError && error.status === 401) {
+        // Retrieve login URL
+        const loginUrl = await client.getOAuthLoginUrl() 
+        
+        // Redirect to Frontegg login page
+        window.location.href = loginUrl;
+        return;
+    }
+    
+    // Rethrow unknown error.
+    throw error
+}
+```
 
-This template represents the culmination of Lokalise technical recommendations as documented in the [Frontend Radar](https://lokalise.atlassian.net/l/cp/Bqkz2hC5).
+You also need to setup OAuth callback path in your browser app.
```

Comment:
maybe add that ideally the `client` instance is shared between these paths so that when `client.fetchAccessTokenByOAuthCode(oauthCode)` is called, it can be reused by the root path. Or, alternatively, we could expand the `/oauth/callback` path to include a call to `const user = await client.getUserData();` (though that seems like a less likely scenario in real-world applications.

Reply:
Updated.